tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
Another use case is reducing boilerplate in pattern matching without introducing intermediate traits:
```scala
sealed trait Tree
class Assign(lhs: Tree, rhs: Tree) extends Tree
class Plus(lhs: Tree, rhs: Tree) extends Tree

(x: Tree) match {
  case x: (Assign | Plus) => foo(x.lhs, x.rhs)
}
```